### PR TITLE
Arduino mega1280 Support

### LINKED ISSUE #1290: Arduino Mega 1280
When I tried to flash: 
````golang
package main

import (
	"machine"
	"time"
)

func main() {
	led := machine.D22

	led.Configure(machine.PinConfig{Mode: machine.PinOutput})

	for {
		println("LOW")
		led.Low()
		time.Sleep(500000000)
		println("HIGH")
		led.High()
		time.Sleep(500000000)
	}
}
````

Using:

````
tinygo flash -target arduino-mega2560 -port /dev/ttyUSB0 blink.go
````
I was getting:

````
	avrdude: Version 6.3, compiled on Jan 28 2020 at 00:00:00
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "/etc/avrdude/avrdude.conf"
         User configuration file is "/home/developer/.avrduderc"
         User configuration file does not exist or is not a regular file, skipping

         Using Port                    : /dev/ttyUSB0
         Using Programmer              : wiring
         Overriding Baud Rate          : 115200
avrdude: stk500v2_ReceiveMessage(): timeout
avrdude: stk500v2_ReceiveMessage(): timeout
avrdude: stk500v2_ReceiveMessage(): timeout
avrdude: stk500v2_ReceiveMessage(): timeout
avrdude: stk500v2_ReceiveMessage(): timeout
avrdude: stk500v2_ReceiveMessage(): timeout
avrdude: stk500v2_getsync(): timeout communicating with programmer

avrdude done.  Thank you.

error: failed to flash /tmp/tinygo654791811/main.hex: exit status 1
````

but Arduino IDE was working, so I debug it to know how Arduino IDE was uploading the .hex file, and I noticed the Arduino IDE is using:
````
avrdude -c arduino -b 57200 -p atmega1280 -P {port} -U flash:w:{hex} -v -D
````

so, I copyed ``arduino-mega2560.json`` and change the flash-command field. and "worked"

basically after this change I can flash but:

- Some analog in pins are inverted, and some are duplicated   
- I can write in digital pins but if I try to read, return false no matter what

### PR DESCRIPTION
**Pull request by MrNinso**
closes #1290

Add Arduino mega 1280 support based in Arduino mega 2560 support.

### CODE REVIEW

File: targets/arduino-mega1280.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "inherits": ["atmega1280"],
+    "build-tags": ["arduino_mega1280"],
+    "ldflags": [
+        "-Wl,--defsym=_bootloader_size=8192"
```
Comment (aykevl) [inline targets/arduino-mega1280.json]: According to the docs it has a bootloader size of 4kB, so this should be:

```suggestion
        "-Wl,--defsym=_bootloader_size=4096"
```

The boards.txt file says the same thing (128kB - 126976 = 4096):

```
mega.menu.cpu.atmega1280.upload.maximum_size=126976
```

File: targets/atmega1280.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "inherits": ["avr"],
+    "cpu": "atmega1280",
+    "build-tags": ["atmega1280", "atmega"],
+    "cflags": [
+        "-mmcu=atmega1280"
+    ],
+    "ldflags": [
+        "-mmcu=avr5",
```
Comment (aykevl) [inline targets/atmega1280.json]: According to https://gcc.gnu.org/onlinedocs/gcc/AVR-Options.html, this should be `-mmcu=avr51`.

**Review by aykevl — COMMENTED** [review]
Overall this looks good to me, with a few small issues.

Can you please also add a smoke test to the Makefile? Something like this:

```Makefile
	$(TINYGO) build -size short -o test.hex -target=arduino-mega1280             examples/blinky1
	@$(MD5SUM) test.hex
```

**Comment by MrNinso** [comment]
Thanks for review @aykevl , I already update the json files and the tests passed

**Comment by deadprogram** [comment]
Thanks for the contribution @MrNinso could you please squash the commits?

**Comment by MrNinso** [comment]
> Thanks for the contribution @MrNinso could you please squash the commits?

Done :+1:

**Comment by deadprogram** [comment]
Thanks @MrNinso now merging.

**Pull request merged**